### PR TITLE
Kernel: Remove trap based syscall handling

### PR DESCRIPTION
**Pull request by agustingianni**
This patch removes the x86 mechanism for calling syscalls, favoring the more modern syscall instruction. It also moves architecture dependent code from functions that are meant to be architecture agnostic therefore paving the way for adding more architectures.

### CODE REVIEW

File: Kernel/Syscall.cpp [hunk 1]
```diff
@@ -138,88 +94,4 @@ ErrorOr<FlatPtr> handle(RegisterState& regs, FlatPtr function, FlatPtr arg1, Fla
...
-    if (result.is_error() && result.error().code() == EPROMISEVIOLATION) {
-        VERIFY(current_thread->is_promise_violation_pending());
-        current_thread->set_promise_violation_pending(false);
-        process.crash(SIGABRT, 0);
-    } else {
-        VERIFY(!current_thread->is_promise_violation_pending());
-    }
-
-    VERIFY(!g_scheduler_lock.is_locked_by_current_processor());
-}
```
Comment (FireFox317) [inline Kernel/Syscall.cpp:223]: Most of this code works fine for aarch64, and should be shared between architectures. I have a patch locally that makes this file work for aarch64 by adding some #ifdefs. The only x86_64 specific things are the `clac()` and the code that adds the random offset to the stack and checks the iopl flags, the rest can and should be shared with aarch64.

Wasn't able to select the whole code block the my phone..

Comment (agustingianni) [reply Kernel/Syscall.cpp:223]: Hello, thanks for the review. Makes sense, I will move this back to its place then and update the PR.

Comment (nico) [reply Kernel/Syscall.cpp:223]: Probably makes sense to wait for #17183 to go in first.

Comment (agustingianni) [reply Kernel/Syscall.cpp]: Hello @FireFox317 and @nico I've rebased the PR and now #17183 is in. Also I've moved back the function that is platform agnostic.

Let me know if there is anything else I can do. Thanks.

File: Kernel/Syscall.cpp [hunk 2]
```diff
@@ -15,60 +16,10 @@
...
-
 namespace Syscall {
 
-static ErrorOr<FlatPtr> handle(RegisterState&, FlatPtr function, FlatPtr arg1, FlatPtr arg2, FlatPtr arg3, FlatPtr arg4);
-
-UNMAP_AFTER_INIT void initialize()
-{
-#if ARCH(X86_64)
-    register_user_callable_interrupt_handler(syscall_vector, syscall_asm_entry);
```
Comment (ADKaster) [inline Kernel/Syscall.cpp:68]: Is this function used anywhere else? if not it should be removed as well in this patch.

Comment (agustingianni) [reply Kernel/Syscall.cpp:68]: Uhm, sorry but this function is not there in the PR, or maybe I'm missing something? You are referring to `initialize` right?

Comment (ADKaster) [reply Kernel/Syscall.cpp:68]: I meant `register_user_callable_interrupt_handler`. It seems useful to check if this was the only call site

Comment (agustingianni) [reply Kernel/Syscall.cpp:68]: Yes, that function is used:

```cpp
    register_user_callable_interrupt_handler(0x01, debug_asm_entry);
    register_user_callable_interrupt_handler(0x03, breakpoint_asm_entry);
```

**Comment by FireFox317** [comment]
Just wondering, but what happens if userspace still executes a `int 0x82`?

**Comment by agustingianni** [comment]
> Just wondering, but what happens if userspace still executes a `int 0x82`?

The code that initialized the int 0x82 handler was executed after the initialization of all the interrupt handlers, so now that we have removed it, the handler registered with `register_interrupt_handler(0x82, interrupt_130_asm_entry);` will be executed.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**